pymdownx/highlight.py

```diff
@@ -52,7 +52,7 @@
         'Default: True'
     ],
     'guess_lang': [
-        False,
+        0,
```

Comment:
I think what would make sense is to set this to `None`. If `None`, we will assume the default of `False`. Otherwise, we will accept booleans or `block` or `inline`. No other strings should be accepted.

Reply:
> No other strings should be accepted.

This makes sense, will reject all other strings not parsed as boolean and not `'inline'` or `'block'`.

> I think what would make sense is to set this to `None`. If `None`, we will assume the default of `False`. Otherwise, we will accept booleans or `block` or `inline`.

I am not sure what benefits would `None` offer here. I think `0` here is functionally equivalent to `False` here except that non-boolean `str` inputs will be allowed? I chose `0` because it is also used in this way by the official `toc` extension [here](https://github.com/Python-Markdown/markdown/blob/a767b2daaad78ba32d45a4f1dabb7c5e218f030a/markdown/extensions/toc.py#L341-L343).

Reply:
> I am not sure what benefits would None offer here. I think 0 here is functionally equivalent to False here except that non-boolean str inputs will be allowed? I chose 0 because it is also used in this way by the official toc extension [here](https://github.com/Python-Markdown/markdown/blob/a767b2daaad78ba32d45a4f1dabb7c5e218f030a/markdown/extensions/toc.py#L341-L343).

I could also show examples in Python Markdown that use None, consider [`codehilite`](https://github.com/Python-Markdown/markdown/blob/a767b2daaad78ba32d45a4f1dabb7c5e218f030a/markdown/extensions/codehilite.py#L280).  Probably more important, we have an example in the very file you are editing [here](https://github.com/facelessuser/pymdown-extensions/blob/main/pymdownx/highlight.py#L72).

Providing a specific type like boolean will cause Python Markdown to type check the options. Why Pyhton Markdown does not type check integers is anyone's guess, but usually, when providing an optional parameter, `None` is [used](https://docs.python.org/3/library/typing.html#typing.Optional), so this makes more sense to me and is my personal preference for this project.

Reply:
It seems that providing `None` only allows a tri-state (`True`/`False`/`None`), but not any other types such as `str`. The relevant snippet is [here](https://github.com/Python-Markdown/markdown/blob/a767b2daaad78ba32d45a4f1dabb7c5e218f030a/markdown/extensions/__init__.py#L60-L61).

The local example of `linenums` is fine because it does not go beyond the three choices (try giving it a string and it will fail in the superclass constructor), and the `codehilite` example gets away by not using the superclass constructor at all.

We can take the latter approach, but it might not be really worth it. Given that `inline` is hardly of any practical use, can we <ins>ab</ins>use the third state of `None` as `block` instead?

Reply:
> It seems that providing None only allows a tri-state (True/False/None), but not any other types such as str. The relevant snippet is [here](https://github.com/Python-Markdown/markdown/blob/a767b2daaad78ba32d45a4f1dabb7c5e218f030a/markdown/extensions/__init__.py#L60-L61).

Ugh, that is awful. Oh well, I guess we leave it as zero.

Reply:
> We can take the latter approach, but it might not be really worth it. Given that inline is hardly of any practical use, can we abuse the third state of None as block instead?

Well, that is a good question. It isn't quite intuitive though. I need to think about this a bit. One option is just to ignore guessing in inline, but I don't know if people actually use guessing for inline...We can probably just use zero as the default, even though I don't like it. Our hands are kind of tied due to the way Python Markdown approaches things like `None`.